### ISSUE TITLE
Scikit-Learn 0.24.* compatibility

### ISSUE THREAD

**Issue by TnTo** [GiulioRossetti/cdlib#165]
**Is your feature request related to a problem? Please describe.**
Actually Scikit-Leran 0.24.* is not supported
The blocking dependency is ASLPAw, which supported Scikit-Learn 0.24.* only from version 2.*.*
ASLPAw relies in its dependency tree on gmpy2.
To be installed with pip gmpy2 requires MPFR, MPC and GMP C libraries installed.

**Describe the solution you'd like**
A conda package is able to ship also the necessary C libraries.
Alternative a warning could be added to the doc and the package. Additionally, ASLPAw could be made an the optional packages.

**Comment by TnTo** [comment]
After some attempts I realized that a conda package needs all its dependencies as conda packages.

Actually, some appear missing:
[ ] angel-cd
[ ] ASLPAw
[ ] demon
[ ] nf1
[ ] omega_index_py3
[ ] eva_lcd
[ ] bimlpa
[ ] markov_clustering
[ ] chinese_whispers
[ ] shuffle_graph
[ ] dynetx
[x] infomap (I've subbitted a package to conda-forge, actually under review)
[ ] thresholdclustering 
[ ] pycombo

Here a first draft for a conda-forge package:
https://github.com/TnTo/staged-recipes/tree/cdlib

**Comment by TnTo** [comment]
FYI @Yquetzal

**Comment by BrandonKMLee** [comment]
Had the same problem bro, thx for discovering

**Comment by GiulioRossetti** [comment]
Hi, 
yes, that was the rationale behind the sklearn version limit in the requirements file.

We already considered the conda-forge alternative, however, since several cdlib dependencies are not conda-packaged  we put it on hold.
We cannot handle continuous up-to-date conda-packaging for third party libraries.

**Comment by BrandonKMLee** [comment]
Observing https://github.com/GiulioRossetti/cdlib/issues/170 Conda does seemed like a good alternative.

**Comment by GiulioRossetti** [comment]
@BrandonKMLee indeed, if it didn't require to have all third-party dependencies conda packaged. 

As mentioned before: we cannot become maintainers of the conda packages for all the libraries we depend on whenever they do not have an official distribution on such channel. It is a time consuming task and we cannot guarantee constant and timely delivering of external packages updates. 

Indeed, if you see this as a valuable path and have time to invest in supporting this feature you are welcome!

**Comment by GiulioRossetti** [comment]
@TnTo I moved ASLPAw to the additional (advanced) requirements - see the repository README. 
Now CDlib supports `scikit-learn>=0.24.1` and requires python>=3.8
